tests/cases/py_client/lib_test.py

```diff
@@ -467,3 +467,56 @@ def _patchRaw(url, request):
 
         # Check we got the right request
         self.assertTrue(patchRequest['valid'])
+
+    def testResourceLookup(self):
+        # Creating item
+        itemName = 'SomethingReallyUnique'
+        item = self.client.createItem(self.publicFolder['_id'],
+                                      itemName)
+
+        testPath = "user/%s/%s/%s" % (self.user['login'],
+                                      self.publicFolder['name'], itemName)
+        testInvalidPath = "user/%s/%s/%s" % (self.user['login'],
+                                             self.publicFolder['name'],
+                                             'RogueOne')
+
+        # Test valid path, default
+        self.assertEqual(self.client.resourceLookup(testPath)['_id'],
+                         item['_id'])
+
+        # Test invalid path, default
+        try:
+            self.client.resourceLookup(testPath)
+        except girder_client.HttpError as e:
+            self.assertEqual(e.status, 400)
+            self.assertEqual(e.method, 'GET')
+            resp = json.loads(e.responseText)
+            self.assertEqual(resp['type'], 'rest')
+            self.assertEqual(resp['message'],
+                             'Path not found: %s' % (testInvalidPath))
```

Comment:
We don't assert that the `except` block was actually entered. Instead, let's do:

```
with self.assertRaises(girder_client.HttpError) as cm:
    self.resourceLookup(testPath)

self.assertEqual(cm.exception.status, 400)
...
```